### PR TITLE
My code review suggestions

### PR DESCRIPTION
**Pull request by Charlie3169**
**Project Description**

Movie Manager allows movie connoisseurs to keep track of movies they watched and stay up to date on what they will watch next. 

* Search for movies
* Add a movie to list
* View movie list in digestible fashion
* Add specific details about movies such as genre. 
* Modify and delete entries as required.
* Mark movie as favorite for easy access
* Add scores to your watched movies and share it with your friends!

**What I learned from this code review:**

-Learned Kotlin naming best practices 
-Learned Kotlin colon conventions for declarations
-Learned Kotlin visibility modifiers and their best practices
https://kotlinlang.org/docs/coding-conventions.html

**Three specific technical suggestions:**
-Are all DTOs that will be created by users able to handle nulls correctly? In many cases a user may not populate all fields, 
or forcing them too could disrupt the user experience.
-Are invalid or illegal actions handled properly? This wasn't a really an issue with the DTO's you started with as they weren't meant for users, but going forward make sure to check user inputs for validity and think about what a user should and shouldn't be able to do. 
-Add JavaDoc describing what is each method and class are doing.

**Changlelog**

**New Files**

_**DTOs**_
-MoviesPlanToWatch
	A mutable set of movies with add and contains methods.
-UserMovieRating
	Has a movie, score, review, and whether or not it is a favorite. Score and review were not explicitly in your project description, but
	felt like natural extensions of it so I put them here. Feel free to remove.
-MoviesWatched
	A mutable set of movies and ratings with add and contains methods for each. Ratings cannot be added to movies that haven't been watched.

_**Unit Tests**_
-MovieListsUnitTest
	Tests MoviesWatched and MoviesPlanToWatch.
-UserMovieRatingUnitTest
	Tests that UserMovieRating can maintain state.
-UserUnitTest
	Tests that User can maintain state.
	

**Other Changes**
User now holds nullable moviesWatched and moviesPlanToWatch fields. There might be other ways of binding these to a user, this was just my approach.

Changed all but ID and username in the User object to be var instead of val, as it is possible in the future that an account management system might be added and in that case it allows for a user to change their password, profileName, and email.

Moved isFavorite parameter on Movies to UserMovieRating, since Movies is something that should be static and stored server side in a database, and Movies are elements in some of these new DTOs, so it wouldn't make sense for those movies to have favorite values.

Added a few function prototypes to MovieDAO and UserDAO that were in line with project requirements.
Added assert statements for MovieUnitTest to encompass all fields.
Renamed main_Activity to mainActivity, just think that is a better name.
Used a with block to add all 10 movies to testMovieData in MainActivity since it looks nicer.

Removed extraneous imports.
Removed every semicolon.
Changed a few .isEquals to == at Android Studio's request.

### CODE REVIEW

**Comment by Charlie3169** [comment]
**Analysis of the program**
Cool idea. Existing code was well structured and the UI looked good as well. Names were descriptive and easily understandable.

**Was the program available in UC Github on time**
Yes it was.

**Is the program documented/commented well enough for you to understand?**
Only two comments that I can see and no documentation, but the program was easy enough to understand regardless.

**Does the program compile?** 
Yes it does.

**Rationale behind your changes.**
Most of my changes were to improve on the core functionality of the project, such as keeping track of movies that haven watched and the scores a user gave them. 
Some refactoring was done to mesh with new DTOs.
The rest of the changes were reformatting things for readability and to be in line with coding conventions.

**Three Technical Concepts I learned**
-Learned about with blocks
-Learned more about making DTOs
-Learned more about making Unit Tests

File: app/src/main/java/com/uc/mymovies/ListAdapter.kt [hunk 1]
```diff
@@ -30,18 +30,18 @@ class ListAdapter(private val context: Context, var movies: ArrayList<Movie>):Ba
             viewHolder = view.tag as ViewHolder
         }
 
-        viewHolder.name?.text = movies[position].title;
-        var imageURL = movies[position].image;
-        Picasso.get().load(imageURL).into(viewHolder.image);
```
Comment (discospiff) [inline app/src/main/java/com/uc/mymovies/ListAdapter.kt:35]: A hard habit to break.

File: app/src/main/java/com/uc/mymovies/MainActivity.kt [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
+            add(movie2)
+            add(movie3)
+            add(movie4)
+            add(movie5)
+            add(movie6)
+            add(movie7)
+            add(movie8)
+            add(movie9)
+            add(movie10)
+        }
```
Comment (discospiff) [inline app/src/main/java/com/uc/mymovies/MainActivity.kt:51]: Good use of Kotlin scoped functions, though I'd go one step further.  Should you be creating/storing data in the Activity?  Probably not.  Refactor this to an external stub class.

File: app/src/test/java/com/uc/mymovies/MovieListsUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    var rating2: UserMovieRating = UserMovieRating(movie5, 10, "This movie is great",true)
+
+    @Test
+    fun moviesWatchedDTO() {
+        givenMoviesWatched()
+        assertTrue(moviesWatched.containsMovie(movie1))
+        assertTrue(moviesWatched.containsMovie(movie2))
+        assertTrue(moviesWatched.containsMovie(movie3))
+        assertTrue(moviesWatched.containsRating(rating1))
+        assertFalse(moviesWatched.containsRating(rating2))
```
Comment (discospiff) [inline app/src/test/java/com/uc/mymovies/MovieListsUnitTest.kt:39]: should follow Given When Then syntax.  Where's the When and Then?

**Comment by discospiff** [comment]
There are a lot of changes here, so I recommend that the team look at them one at a time, and decide what to merge.

Several good suggestions.

I generally shy away from adding new classes/new functionality as part of a code review.  Code reviews should focus on reducing technical debt of existing source code.  One exception is unit tests, provided that the unit tests increase test coverage.  If that is the case, I'd update the unit test to use Given When Then syntax, then consider using them.

File: app/src/main/java/com/uc/mymovies/ListAdapter.kt (hunk 1 above)
Comment (helloseth) [reply app/src/main/java/com/uc/mymovies/ListAdapter.kt:35]: I do a lot of reactjs and often always leave them out. Definitely a habit I need to break!

**Pull request closed without merging**